src/FIO/Signer.cpp

```diff
@@ -0,0 +1,97 @@
+// Copyright © 2017-2020 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Signer.h"
+#include "Address.h"
+#include "../Base58.h"
+#include "../Hash.h"
+#include "../HexCoding.h"
+
+#include <TrezorCrypto/ecdsa.h>
+#include <TrezorCrypto/secp256k1.h>
+
+#include <cassert>
+
+namespace TW::FIO {
+
+using namespace std;
+
+Data Signer::sign(const PrivateKey& privKey, const Data& data) {
+    Data hash = Hash::sha256(data);
+    return signData(privKey, hash);
+}
+
+Data Signer::signData(const PrivateKey& privKey, const Data& data) {
+    Data signature = privKey.sign(data, TWCurveSECP256k1, is_canonical);
+    return signature;
+}
+
+std::string Signer::signatureToBsase58(const Data& sig) {
+    Data sigWithSuffix(sig);
+    append(sigWithSuffix, TW::data(SignatureSuffix));
+    // take hash, ripemd, first 4 bytes 
+    Data hash = Hash::ripemd(sigWithSuffix);
+    Data sigWithChecksum(sig);
+    append(sigWithChecksum, TW::data(hash.data(), 4));
+    string s = SignaturePrefix + Base58::bitcoin.encode(sigWithChecksum);
+    return s;
+}
+
+bool Signer::verify(const PublicKey& pubKey, const Data& data, const Data& signature) {
+    return (0 == ecdsa_verify_digest(&secp256k1, pubKey.bytes.data(), signature.data() + 1, data.data()));
```

Comment:
can we use `PublicKey::verify`?

Reply:
Yes, it can, thanks.  Though first byte of signature has to be discarded first.